scala_proto/private/compile_proto.bzl

```diff
@@ -0,0 +1,82 @@
+load("//scala/private:rule_impls.bzl", "compile_scala")
+load("//scala/private:common.bzl", "write_manifest_file")
+load("//scala/private:dependency.bzl", "legacy_unclear_dependency_info_for_protobuf_scrooge")
+
+def _concat_lists(list1, list2):
+    all_providers = []
+    all_providers.extend(list1)
+    all_providers.extend(list2)
+    return all_providers
+
+def compiled_jar_file(actions, scalapb_jar):
+    scalapb_jar_name = scalapb_jar.basename
+
+    # ends with .srcjar, so remove last 6 characters
+    without_suffix = scalapb_jar_name[0:len(scalapb_jar_name) - 6]
+
+    # this already ends with _scalapb because that is how scalapb_jar is named
+    compiled_jar = without_suffix + "jar"
+    return actions.declare_file(compiled_jar, sibling = scalapb_jar)
+
+def compile_proto(
```

Comment:
this isn't really compiling proto, this is compiling scala code. The scala has been generated at this point.